src/path/sentry_path_unix.c

```diff
@@ -156,23 +156,19 @@ sentry_path_t *
 sentry__path_from_str(const char *s)
 {
     char *path = sentry__string_clone(s);
-    sentry_path_t *rv = NULL;
     if (!path) {
         return NULL;
     }
-    rv = sentry__path_from_str_owned(path);
-    if (rv) {
-        return rv;
-    }
-    sentry_free(path);
-    return NULL;
+    // NOTE: function will free `path` an error
```

Comment:
That doesn't really read like english... "function takes ownership of `path`, meaning future `free(path)` calls would be an error" is what is meant?

Reply:
`on error` ;-) Meaning yes, it will take ownership and free if necessary.